opengrok-indexer/src/main/java/org/opengrok/indexer/util/TandemFilename.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opengrok.indexer.util;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+
+/**
+ * Represents a utility class for creating a filename to operate in tandem with
+ * an original filename by adding a new file extension but limiting the length
+ * of the new filename to 255 UTF-8 encoded bytes if necessary by truncating
+ * and packing in a Base64-encoded SHA-256 hash of the original file name.
+ */
+public class TandemFilename {
+
+    private static final int MAX_BYTES = 255;
+
+    /**
+     * One fewer than {@link #MAX_BYTES} as a cap for simple concatenation to
```

Comment:
could you expand on the collision ?

Reply:
Trivial collision is prevented by disallowing simple concatenation to `MAX_BYTES`.

E.g.  in a simplified example, if `MAX_BYTES` were 16 and a base64 digest were five bytes long, then for a short name, `filename1`, a tandem filename for xref would be `filename1.gz`. For a long name, `filename0123456789`, a tandem filename for xref might be `filename5cdb0.gz`

If simple concatenation were allowed to 16 bytes, then one could drop in a file named `filename5cdb0`, and OpenGrok would concatenate for a tandem filename for xref as `filename5cdb0.gz` and collide with the long file above.

Reply:
I see, that makes sense; I like the robustness.